tests: drivers: watchdog testing on stm32 iwdg and wwdg

FRASTM opened the pull request:
Configure the testcase.yaml to execute watchdog testcase
wdt_basic_api on each IWDG and WWDG on all the stm32 boards
With two generic stm32 .overlay files as extra config :
one wdg is tested when the other is disabled.
Removing from board overlay.
Removing the boards/nucleo_g431.overlay : target not available

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/45691

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
tests: drivers: watchdog testing on stm32 iwdg and wwdg

Configure the testcase.yaml to execute watchdog testcase
wdt_basic_api on each IWDG and WWDG on all the stm32 boards
With two generic stm32 .overlay files as extra config :
one wdg is tested when the other is disabled.
Removing from board overlay.
Giving the list of boards else non-stm32 ones could try building

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
dts: arm: stm32l5 Invalid wwdg interrupt priority

Set the priority of the wwdg instance to 6
else a build error happens :
static assertion failed: "Invalid interrupt priority.
Values must not exceed IRQ_PRIO_LOWEST"

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on tests/drivers/watchdog/wdt_basic_api/testcase.yaml:
```diff
@@ -1,21 +1,20 @@
-    depends_on: watchdog
-    tags: drivers watchdog
-    filter: dt_compat_enabled("st,stm32-window-watchdog")
+    filter: dt_compat_enabled("st,stm32-window-watchdog") or dt_compat_enabled("st,stm32-window-watchdog")
+    extra_args: DTC_OVERLAY_FILE="boards/stm32_wwdg.overlay"
```
This filter seems totally fine to me and yet, it doesn't have the expected effect, as twister still atempts to build it on non complying boards:
```
INFO    - 130/584 frdm_k22f                 tests/drivers/watchdog/wdt_basic_api/drivers.watchdog.stm32iwdg FAILED Cmake build failure (build)
INFO    - /__w/zephyr/zephyr/twister-out/frdm_k22f/tests/drivers/watchdog/wdt_basic_api/drivers.watchdog.stm32iwdg/build.log
ERROR   - Loading Zephyr default modules (Zephyr base).
-- Application: /__w/zephyr/zephyr/tests/drivers/watchdog/wdt_basic_api
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter 
-- Cache files will be written to: /__w/zephyr/zephyr/.cache
-- Zephyr version: 3.1.0-rc1 (/__w/zephyr/zephyr)
-- Found west (found suitable version "0.13.0", minimum required is "0.7.1")
-- Board: frdm_k22f
-- Found host-tools: zephyr 0.14.1 (/opt/toolchains/zephyr-sdk-0.14.1)
-- Found dtc: /opt/toolchains/zephyr-sdk-0.14.1/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found toolchain: zephyr 0.14.1 (/opt/toolchains/zephyr-sdk-0.14.1)
-- Found BOARD.dts: /__w/zephyr/zephyr/boards/arm/frdm_k22f/frdm_k22f.dts
-- Found devicetree overlay: boards/stm32_iwdg.overlay
devicetree error: ./boards/stm32_iwdg.overlay:7 (column 1): parse error: undefined node label 'wwdg'
```

FRASTM commented:
list the stm32 target boards to avoid compilation error on Non-stm32 
add the stm32u5 where the wWDG is instance wwdg1
add the stm32h7 where the iWDG is instance iwdg1

erwango commented on tests/drivers/watchdog/wdt_basic_api/boards/stm32_wwdg_h7.overlay:
```diff
@@ -0,0 +1,13 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&wwdg1 {
```
In order to simplify things, I'd propose to update stm32h7.dtsi definition with:
```
		wwdg: wwdg1: watchdog@50003000 {
```
This will enable use of wwdg on this series as well.
Same for iwdg.

FRASTM committed:
dts: arm: stm32h7 declare wwdg and iwdg node instances

Overload the IWDG and WWDG first instance declaration
to align with other stm32 mcus.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
dts: arm: stm32u5 and mp1 declare wwdg node instances

Overload the IWDG first instance declaration
to align with other stm32 mcus.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
rebase on 903672c394f0102059ed07fc2a2831bbe66a2f30
